src/components/editor-page/editor-pane/max-length-warning/max-length-warning.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { MaxLengthWarningModal } from './max-length-warning-modal'
+import { useNoteMarkdownContent } from '../../../../hooks/common/use-note-markdown-content'
+import { useApplicationState } from '../../../../hooks/common/use-application-state'
+
+/**
+ * Watches the length of the document and shows a warning modal to the user if the document length exceeds the configured value.
+ */
+export const MaxLengthWarning: React.FC = () => {
+  const [showMaxLengthWarning, setShowModal] = useState(false)
```

Comment:
Use consistent names

```suggestion
  const [showMaxLengthWarning, setShowMaxLengthWarning] = useState(false)
```